index.html

```diff
@@ -5099,6 +5125,31 @@ <h2 id="defining-a-valid-audioworkletprcoessor">
                 <code>"name"</code>.
               </li>
             </ol>
+            <p>
+              The return value of this method controls the lifetime of the
+              <a>AudioWorkletProcessor</a>'s associated <a>AudioWorkletNode</a>
+              by determining the existence of the node's <a href=
+              "#dfn-active-reference">active reference</a>. At the conclusion
+              of each call to the <code>process()</code> method, if the result
+              of applying <a href=
+              "https://tc39.github.io/ecma262/#sec-toboolean"><code>ToBoolean</code></a>
+              (described in [[!ECMASCRIPT]]) to the return value is equal to
+              <code>true</code>, then a single <a>active reference</a> is
+              retained to the node, affecting the node's <a href=
+              "#lifetime-AudioNode">lifetime</a>. Otherwise, any active
+              reference is removed and the node's lifetime depends only on
+              other references as they may exist.
+            </p>
+            <div class="note">
+              While returning <code>true</code> ensures that
+              <code>process()</code> will be called during subsequent rendering
+              quanta, returning <code>false</code> does not prevent future
+              calls to <code>process()</code>. Other references to the
+              corresponding <a>AudioWorkletNode</a> may exist and the node may
+              thus continue to function. Furthermore, these future calls may
+              return a value of <code>true</code>, reinstating the active
+              reference.
```

Comment:
The text says this (emphasis added): 

> Furthermore, these future calls may  return a value of <code>true</code>, _reinstating the active  reference_.

So what gets reinstated is merely the active reference, not the node. And your description is correct: if there was a JS reference, then returning false never pulled the node off the graph. As a corollary, returning true merely reinstated the active reference (as the text says), not the node.

Going on from this, I am not saying the UA defers paying attention to the return value of process(). What I said is, the UA immediately honors the return value by creating/destroying the active reference. The part that can be deferred is the UA _noticing that there are no more references_. And I think allowing it to be deferred might be a good thing, because otherwise the information about whether JS references exist needs to be made synchronously available on the rendering thread.